### PR TITLE
Intl.DurationFormat update coverage

### PR DESCRIPTION
**Pull request by romulocintra**
Progressively adding a few more tests to Intl.Duration Format to improve ctor coverage
 
cc @sffc 

#3305

### CODE REVIEW

File: test/intl402/DurationFormat/constructor/constructor/locales-invalid.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// Copyright 2022 Igalia, S.L. All rights reserved.
```
Comment (ptomato) [inline test/intl402/DurationFormat/constructor/constructor/locales-invalid.js:1]: Looking at the structure of existing tests elsewhere in Intl, e.g. for DateTimeFormat, it seems like these tests should go directly in the `test/intl402/DurationFormat` directory, without the `constructor/constructor/`. See the thread https://github.com/tc39/test262/pull/3406#discussion_r809537160

File: test/intl402/DurationFormat/constructor/constructor/options-localeMatcher-valid.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+const validOptions = [
+  undefined,
+  'lookup',
+  'best fit'
+];
+
+for (const localeMatcher of validOptions) {
+  const obj = new Intl.DurationFormat('en', { localeMatcher });
+
+  assert.sameValue(Object.getPrototypeOf(obj), Intl.DurationFormat.prototype, `proto check - ${localeMatcher}`);
```
Comment (ptomato) [inline test/intl402/DurationFormat/constructor/constructor/options-localeMatcher-valid.js:23]: I'm guessing this is a stand-in for lack of a better assertion that says "constructing the object didn't throw"?
Is it possible to check this option via `obj.resolvedOptions()` instead?

Comment (ptomato) [inline test/intl402/DurationFormat/constructor/constructor/options-localeMatcher-valid.js:23]: I'm guessing this is a stand-in for lack of a better assertion that says "constructing the object didn't throw"?

File: test/intl402/DurationFormat/constructor/supportedLocalesOf/basic.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+assert.sameValue(typeof Intl.DurationFormat.supportedLocalesOf, "function",
+                 "supportedLocalesOf should be supported.");
+
+const defaultLocale = new Intl.DurationFormat().resolvedOptions().locale;
+const notSupported = "zxx"; // "no linguistic content"
+const requestedLocales = [defaultLocale, notSupported];
+
+const supportedLocales = Intl.DurationFormat.supportedLocalesOf(requestedLocales);
+assert.sameValue(supportedLocales.length, 1, "The length of supported locales list is not 1.");
```
Comment (ptomato) [inline test/intl402/DurationFormat/constructor/supportedLocalesOf/basic.js:18]: These assertion messages seem unusual compared to what I've seen elsewhere in test262. I think "The length of the supported locales list should be 1" would be more usual. Same for the message just below this one.

**Review by ptomato — COMMENTED** [review]
Thanks!

File: test/intl402/DurationFormat/constructor-options-order.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    actual.push("style");
+    return undefined;
+  },
+};
+
+const expected = [
+  "localeMatcher",
+  "numberingSystem",
+  "style"
+];
```
Comment (ptomato) [inline test/intl402/DurationFormat/constructor-options-order.js:38]: Optional: There is a helper for testing the order of operations of property gets, although maybe it should get a more general place than TemporalHelpers if we're going to use it here as well.

```suggestion
const options = {};
TemporalHelpers.observeProperty(actual, options, "localeMatcher", undefined);
TemporalHelpers.observeProperty(actual, options, "numberingSystem", undefined);
TemporalHelpers.observeProperty(actual, options, "style", undefined);

const expected = [
  "get localeMatcher",
  "get numberingSystem",
  "get style"
];
```

File: test/intl402/DurationFormat/supportedLocalesOf/basic.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+assert.sameValue(typeof Intl.DurationFormat.supportedLocalesOf, "function",
+                 "supportedLocalesOf should be supported.");
+
+const defaultLocale = new Intl.DurationFormat().resolvedOptions().locale;
+const notSupported = "zxx"; // "no linguistic content"
+const requestedLocales = [defaultLocale, notSupported];
+
+const supportedLocales = Intl.DurationFormat.supportedLocalesOf(requestedLocales);
+assert.sameValue(supportedLocales.length, 1, "The length of the supported locales list should be 1");
+assert.sameValue(supportedLocales[0], defaultLocale, "The default locale is not returned in the supported list.");
```
Comment (ptomato) [inline test/intl402/DurationFormat/supportedLocalesOf/basic.js:19]:
```suggestion
assert.sameValue(supportedLocales[0], defaultLocale, "The default locale is returned in the supported list.");
```
?

**Review by ptomato — DISMISSED** [review]
LGTM, one optional improvement (up to you to decide) and one wording nitpick

File: test/intl402/DurationFormat/constructor-options-order.js (hunk 4 above)
Comment (romulocintra) [reply test/intl402/DurationFormat/constructor-options-order.js:38]: That's nice, I'm ok with using it, no strong preference at moment, only  the "TemporalHelpers" name, would be really nice to have it as generic utils functions and documented somewhere else cause they're really handy

**Review by ptomato — APPROVED** [review]

**Pull request merged**